pod.go

```diff
@@ -774,6 +776,45 @@ func (p *Pod) startVM(netNsPath string) error {
 	return nil
 }
 
+func (p *Pod) createProxyProcess() error {
+	pid, uri, err := p.proxy.start(*p)
+	if err != nil {
+		return err
+	}
+
+	// save state
+	p.state.URL = uri
+	if err := p.setPodState(p.state); err != nil {
+		return err
+	}
+
+	p.proxyProcess = newProcess("", pid)
+
+	p.Logger().WithField("proxy-pid", pid).Debug("proxy started")
+
+	return nil
+}
+
+// waitForProxy waits for the proxy process to finish
+func (p *Pod) waitForProxy() error {
+	pid := p.proxyProcess.Pid
```

Comment:
I don't see any place where you save this info about the proxy PID. When `DeletePod()` will be called, I think that `p.proxyProcess` is going to be empty.

Reply:
Good catch - fixing...